test/spec/modules/mwOpenLinkIdSystem_spec.js

```diff
@@ -0,0 +1,25 @@
+import { writeCookie, mwOpenLinkIdSubModule } from 'modules/mwOpenLinkIdSystem.js';
+
+const P_CONFIG_MOCK = {
+  params: {
+    accountId: '123',
+    partnerId: '123'
+  }
+};
+
+describe('mwOpenLinkId module', function () {
+  beforeEach(function() {
+    writeCookie('');
+  });
+
+  it('getId() should return a MediaWallah openLink Id when the MediaWallah openLink first party cookie exists', function () {
+    writeCookie({eid: 'XX-YY-ZZ-123'});
+    const id = mwOpenLinkIdSubModule.getId(P_CONFIG_MOCK);
+    expect(id).to.be.deep.equal({id: {eid: 'XX-YY-ZZ-123'}});
+  });
+
+  it('getId() should return an empty Id when the MediaWallah openLink first party cookie is missing', function () {
```

Comment:
I don't understand this behavior. If you have a new user that you haven't seen before (i.e. no 1p cookie), you generate an ID in the `SetID` method, so why would you not use that? In the current code (excluding my suggestion above), you would create a new ID and store it in a cookie, but not use it on that first request...which means you're module doesn't work until the second page view for a user on the same site. Is that intended?

Reply:
Done!